src/twisted/web/test/test_web.py

```diff
@@ -1849,3 +1852,50 @@ def test_defaultReactor(self):
 
         factory = http.HTTPFactory()
         self.assertIs(factory.reactor, reactor)
+
+
+class QueueResource(Resource):
```

Comment:
why is this test added to test_web.py and not to test_http.py ?

Just asking.

Reply:
`test_http` already has a test for this which doesn't cover the full stack top-to-bottom.

The boundaries are fuzzy, and this fix obviously has its foot in both worlds, but HTTP is for the HTTP protocol parsing, whereas web is for request handling.  This is for the boundary where a parsed HTTP protocol message *becomes* an invocation of the request handling logic.

At first I felt like the problem might be at a higher level because there's already a test for this in `test_http`, before I realized that it was a TCP segment colocality issue.  But, problems *might* be at a higher level in the future, so having the test live at this later, both in `test_web` and with `server.Site` being the layer of the public API getting tested, still makes sense to me.

Reply:
Maybe test_http.py needs a docstring that is more than `Test HTTP support.` same for `test_web.py` with `Tests for various parts of L{twisted.web}.` 

I will keep this in mind and will try to update the docstring, next time I will touch these files